### PR TITLE
Shultsedu 1489

### PR DESCRIPTION
**Pull request by Nastyshka14**

### CODE REVIEW

File: src/components/Calendar/CalendarPage.tsx [hunk 1]
```diff
@@ -1,204 +1,68 @@
...
-    window.open(navUrl, '_blank')
+  const content = (value: Moment): JSX.Element => {
+    const listData: IBirthdayItem[] = getListData(value)
+    return (
+      <ul className='content'>
+        {listData.length
+          ? listData.map((item: IBirthdayItem): JSX.Element => (
+              <div className='content__component'>
+                <li className='content__item' key={item.name}>
+                  <div className='content__item_name'>
```
Comment (PodshyvalovaRuslana) [inline src/components/Calendar/CalendarPage.tsx]: it's better to give the names for elements with dashes between words to make them more readable, in this case will be .content__item-name

File: src/components/Calendar/CalendarPage.tsx [hunk 2]
```diff
@@ -1,204 +1,68 @@
...
-              </li>
-            ))
-          : ''}
-      </ul>
+      <Popover title={title(value)} placement='right' content={content(value)} trigger='click'>
+        <ul className='events'>
+          {listData.length
+            ? listData.map((item: IBirthdayItem):JSX.Element => (
+                <div>
+                  <li key={item.name}>
```
Comment (PodshyvalovaRuslana) [inline src/components/Calendar/CalendarPage.tsx:58]: please give more unique value for key

File: src/components/Sprites/Sprites.tsx [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+  function shareOnEmail(item: string) {
+    const navUrl = `mailto:?subject=Today is ${item}`
+    window.open(navUrl, '_blank')
+  }
+
+  function shareOnSMS(item: string) {
+    const navUrl = `sms:?body=Today is ${item}`
+    window.open(navUrl, '_blank')
+  }
+
```
Comment (PodshyvalovaRuslana) [inline src/components/Sprites/Sprites.tsx]: please change the logic of component to avoid multiple function creation

**Review by PodshyvalovaRuslana — CHANGES_REQUESTED** [review]

File: src/components/Sprites/Sprites.tsx [hunk 4]
```diff
@@ -2,69 +2,49 @@ import React from 'react'
...
-    const navUrl = `mailto:?subject=Today is ${item}`
-    window.open(navUrl, '_blank')
-  }
+  
+  const vkUrl = `http://vk.com/share.php?url=${window.location.href}&title=Today is ${item}`
+  const telegramUrl = `https://t.me/share/url?url=${window.location.href}&text=Today is ${encodeURIComponent(item)}`
+  const viberUrl = `viber://forward?text=Today is ${item}`
+  const whatsappUrl = `whatsapp://send?text=Today is ${item}`
+  const emailUrl = `mailto:?subject=Today is ${item}`
+  const smsUrl = `sms:?body=Today is ${item}`
```
Comment (PodshyvalovaRuslana) [inline src/components/Sprites/Sprites.tsx]: you also can make it like an object instead of bunch of constants, for make it more readable

**Review by PodshyvalovaRuslana — APPROVED** [review]

File: cordovaApp/www/index.js [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+// Wait for the deviceready event before using any of Cordova's device APIs.
+// See https://cordova.apache.org/docs/en/latest/cordova/events/events.html#deviceready
+
+document.addEventListener('deviceready', onDeviceReady, false);
+
+function onDeviceReady() {
+    // Cordova is now initialized. Have fun!
+
+    console.log('Running cordova-' + cordova.platformId + '@' + cordova.version);
```
Comment (s-severin) [inline cordovaApp/www/index.js]: Remove console.log

File: cordovaApp/www/js/index.js [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+ */
+
+// Wait for the deviceready event before using any of Cordova's device APIs.
+// See https://cordova.apache.org/docs/en/latest/cordova/events/events.html#deviceready
+document.addEventListener('deviceready', onDeviceReady, false);
+
+function onDeviceReady() {
+    // Cordova is now initialized. Have fun!
+
+    console.log('Running cordova-' + cordova.platformId + '@' + cordova.version);
```
Comment (s-severin) [inline cordovaApp/www/js/index.js]: Remove console.log

File: public/index.js [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+
+// Wait for the deviceready event before using any of Cordova's device APIs.
+// See https://cordova.apache.org/docs/en/latest/cordova/events/events.html#deviceready
+
+document.addEventListener('deviceready', onDeviceReady, false);
+
+function onDeviceReady() {
+    // Cordova is now initialized. Have fun!
+
+    console.log('Running cordova-' + cordova.platformId + '@' + cordova.version);
```
Comment (s-severin) [inline public/index.js]: Remove

File: src/api/utils/getBirthdaysData.ts [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+    [newBirthdaysList, openNotification],
+  )
+
+  const getReminder = useCallback(() => {
+    const todayBirthdaysList = JSON.parse(localStorage.getItem('todayBirthdaysList'))
+    birthdays.forEach((item: IBirthdayItem) => {
+      if (valueToString(item.date) === valueToString(moment())) {
+        if (todayBirthdaysList !== null) {
+          if (
+            todayBirthdaysList.find((element: IBirthdayItem) => {
```
Comment (s-severin) [inline src/api/utils/getBirthdaysData.ts]: Move to variable

File: src/api/utils/getBirthdaysData.ts [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+              return element.name !== item.name && element.date !== item.date
```
Comment (s-severin) [inline src/api/utils/getBirthdaysData.ts]: todayBirthdaysList.find((element: IBirthdayItem) => element.name !== item.name && element.date !== item.date

Comment (s-severin) [reply src/api/utils/getBirthdaysData.ts]: И деструктуризируй для читаемости

File: src/components/Calendar/CalendarPage.scss [hunk 10]
```diff
@@ -2,95 +2,49 @@
...
-
-  &__viber {
-    background: url(icons/viber.svg);
-  }
-
-  &__whatsapp {
-    background: url(icons/whatsapp.svg);
-  }
+.title__day {
+  height: 1.4em;
```
Comment (s-severin) [inline src/components/Calendar/CalendarPage.scss]: Why `em`?

File: src/components/Notification/Notification.scss [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+}
+
+.message {
+  display: flex;
+  align-items: flex-end;
+  border-bottom: 1px solid #0050b3;
+}
+
+.message__text {
+  height: 1.4em;
```
Comment (s-severin) [inline src/components/Notification/Notification.scss]: Use px

File: src/hooks/useContentful.ts [hunk 12]
```diff
@@ -1,29 +1,34 @@
...
       if (body) {
-        body = JSON.stringify(body);
+        body = JSON.stringify(body)
       }
       const response = await fetch(apiUrl, { method: 'POST', body, headers })
       if (!response.ok) {
-        throw new Error(response.statusText || 'Something went wrong');
+        throw new Error(response.statusText || 'Something went wrong')
       }
       const data = response.text()
```
Comment (s-severin) [inline src/hooks/useContentful.ts]: `return response.text()`

**Review by s-severin — CHANGES_REQUESTED** [review]

**Review by s-severin — APPROVED** [review]

**Pull request merged**